### PR TITLE
logging_decorator was added

### PR DESCRIPTION
**Pull request by borismod**
## Description
Logging decorator was added 

## Related Stories
None

## Breaking
NO

---
This change is [Reviewable](https://reviewable.io/reviews/qualisystems/vcentershell/653)

### CODE REVIEW

**Comment by doppleware** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[vcentershell_driver/driver.py, line 73 \[r1\]](https://reviewable.io:443/reviews/qualisystems/vcentershell/653#-KFcWRKE8fT_QURN-U3H:-KFcWRKE8fT_QURN-U3I:-910594072) ([raw file](https://github.com/qualisystems/vcentershell/blob/8d87f2dcbdb73bbb120d1abb896d0745e7551dc9/vcentershell_driver/driver.py#L73)):*
So in general I thought this should be more like 'DriverCommandDecorator' and then we can configure it by default to have logging or other functions. And then what we need is this 'composite' decorator class that can be easily configured with different interceptors like logging, vcenter connection etc.

---

*[vcentershell_driver/logging_decorator.py, line 18 \[r1\]](https://reviewable.io:443/reviews/qualisystems/vcentershell/653#-KFcVwmYS45MZHGfUVb2:-KFcVwmYS45MZHGfUVb3:-337664527) ([raw file](https://github.com/qualisystems/vcentershell/blob/8d87f2dcbdb73bbb120d1abb896d0745e7551dc9/vcentershell_driver/logging_decorator.py#L18)):*
Instead of checking the name of the arg maybe we can check the type? If its class name is one of our context classes?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qualisystems/vcentershell/653)*

**Comment by borismod** [comment]
@doppleware can you review please?

**Comment by borismod** [comment]
@doppleware can you review please?

File: package/cloudshell/cp/vcenter/commands/cloudshell_session_factory.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    def context_ended(self):
+        pass
+
+    def __enter__(self):
+        self.context_started()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.context_ended()
+
+
```
Comment (doppleware) [inline package/cloudshell/cp/vcenter/commands/cloudshell_session_factory.py]: This class should be moved to cloudshell-automation-api

**Pull request closed without merging**